### ISSUE TITLE
Charts initialized with an empty dataset cause errors on change detection

### ISSUE THREAD

**Issue by danmana** [valor-software/ng2-charts#1265]
**Reproduction of the problem**
1. Initialize a chart with `datasets=[]` empty array (ex: when the user has to select the datasets and there are none selected by default)
2. Trigger change detection (ex: hover over chart)

Result: Errors show up in the console.

```
ERROR TypeError: Cannot set property 'data' of undefined
    at BaseChartDirective.propagateDataToDatasets (valor-software-ng2-charts.umd.js? [sm]:1138)
    at BaseChartDirective.ngDoCheck (valor-software-ng2-charts.umd.js? [sm]:652)
```

If later a dataset is added, the errors go away.
And if we remove again all datasets the errors don't come back. Only when the chart is initialized with an empty datasets array.

Demo: https://stackblitz.com/edit/angular-ivy-c8oztr?file=src%2Fapp%2Fapp.component.ts

The problem is caused by this code from [propagateDataToDatasets](https://github.com/valor-software/ng2-charts/blob/development/projects/ng2-charts/src/lib/base-chart.directive.ts#L497), which assumes that if datasets is defined, there is at least one dataset (see else block).
```js
      if (!this.datasets) {
        this.datasets = [{ data: newDataValues }];
        if (this.chart) {
          this.chart.data.datasets = this.datasets;
        }
      } else {
        this.datasets[0].data = newDataValues;
        this.datasets.splice(1); // Remove all elements but the first
      }
```

**Comment by paviad** [comment]
Fix incorporated into version 2.4.2

**Comment by hudzenko** [comment]
Hi there, we experience now weird behaviour when [datasets]="[]". baseChart code somehow replaces objects and inputs in just totally other part of applications. Mostly it somehow replaces angular Inputs with the chart related object for totally different components of our app

This is super weird..
